hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -1010,6 +1014,23 @@ public Data executeOnKeyInternal(Data key, EntryProcessor entryProcessor) {
         }
     }
 
+    private static void validateEntryProcessorForSingleKeyProcessing(EntryProcessor entryProcessor) {
+        if (entryProcessor instanceof Offloadable) {
```

Comment:
I think you should swap the places of if (entryProcessor instanceof Offloadable) {... and if (entryProcessor instanceof ReadOnly) {... blocks.